components/cluster.py

```diff
@@ -200,8 +259,11 @@ def compute_pos(df, nodes):
     for n in nodes:
         row = df[df["EVENTID"] == int(n)]
         x = row["DateTime"].values[0]
+        x = dt.fromtimestamp(x//10**9)
         y = row["MAGNITUDE"].values[0]
-        pos[n] = (x, y)
+        lat = row["LATITUDE"].values[0]
+        lon = row["LONGITUDE"].values[0]
+        pos[n] = (x, y, lat, lon)
```

Comment:
Not sure about this, but it might be faster to iterate over each row in the data frame with `for _, row in df.iterrrows()` as now you loop over each eventid and then search each one separately from the data frame

Reply:
I think here I'm not iterating over the rows in the dataset but  but over the nodes in a graph. And then finding the corresponding row in the dataset.

Reply:
Ah, you're right! Never mind me!